apps/google-analytics-4/frontend/src/helpers/contentfulContext.ts

```diff
@@ -0,0 +1,31 @@
+import { ContentfulContext } from '../types';
+import { KnownSDK, locations } from '@contentful/app-sdk';
+
+const Locations = [
+  locations.LOCATION_APP_CONFIG,
+  locations.LOCATION_ENTRY_FIELD,
+  locations.LOCATION_ENTRY_EDITOR,
+  locations.LOCATION_DIALOG,
+  locations.LOCATION_ENTRY_SIDEBAR,
+  locations.LOCATION_PAGE,
+  locations.LOCATION_HOME,
+];
+
+export const location = (sdk: KnownSDK) => {
+  let currentLocation = undefined;
+
+  Locations.forEach((location) => {
+    if (sdk.location.is(location)) {
+      currentLocation = location;
```

Comment:
Maybe just `return location` to terminate early on a match?

Reply:
ah good catch, updating now